### PR TITLE
add a TryReadLock function

### PR DESCRIPTION
**Pull request by natebrennand**
ticket - https://clever.atlassian.net/browse/INFRA-3260

### CODE REVIEW

File: rw_mutex.go [hunk 1]
```diff
@@ -101,24 +104,52 @@ func (m *RWMutex) RLock() error {
...
+			jitter := time.Duration(rand.Int63n(1000)) * time.Millisecond
+			time.Sleep(m.SleepTime + jitter)
+			continue
+		}
+		return err
+	}
+}
+
+// TryReadLock tries to acquires the read lock
+func (m *RWMutex) TryReadLock() error {
```
Comment (meliaj) [inline rw_mutex.go]: Not sure if we should stick to convention and call this `TryRLock`?

File: rw_mutex.go [hunk 2]
```diff
@@ -101,24 +104,52 @@ func (m *RWMutex) RLock() error {
 	}
```
Comment (meliaj) [inline rw_mutex.go]: Can we remove this loop above since it's in `tryToGetReadLock`?

**Review by meliaj — APPROVED** [review]
LGTM

File: rw_mutex.go (hunk 1 above)
Comment (natebrennand) [reply rw_mutex.go]: 19f20b9f2dd5c590d987da2e539a30c99fc504c6

File: rw_mutex.go (hunk 2 above)
Comment (natebrennand) [reply rw_mutex.go]: good catch
0aa9532b46656656ffa10b5e2ab47a1a266f8b8d

**Pull request merged**